File: Programas Python/Calculadora de Áreas y Perímetros/Calculadora de Áreas y Perímetros.py
import math

print("Bienvenido a la calculadora geométrica")
while True:
    print("\nOpciones:")
    print("1. Círculo")
    print("2. Rectángulo")
    print("3. Triángulo")
    print("4. Salir")
    opcion = input("Selecciona una figura (1-4): ")

    if opcion == '4':
        print("¡Hasta luego!")
        break

    if opcion == '1':
        try:
            radio = float(input("Radio del círculo: "))
            area = math.pi * radio ** 2
            perimetro = 2 * math.pi * radio
            print(f"Área: {area:.2f}, Perímetro: {perimetro:.2f}")
        except ValueError:
            print("Error: Ingresa un valor numérico válido.")

    elif opcion == '2':
        try:
            base = float(input("Base del rectángulo: "))
            altura = float(input("Altura del rectángulo: "))
            area = base * altura
            perimetro = 2 * (base + altura)
            print(f"Área: {area:.2f}, Perímetro: {perimetro:.2f}")
        except ValueError:
            print("Error: Ingresa valores numéricos válidos.")

    elif opcion == '3':
        try:
            a = float(input("Lado 1 del triángulo: "))
            b = float(input("Lado 2 del triángulo: "))
            c = float(input("Lado 3 del triángulo: "))
            if a + b > c and a + c > b and b + c > a:
                perimetro = a + b + c
                s = perimetro / 2
                area = math.sqrt(s * (s - a) * (s - b) * (s - c))
                print(f"Área: {area:.2f}, Perímetro: {perimetro:.2f}")
            else:
                print("Error: Los lados no forman un triángulo válido.")
        except ValueError:
            print("Error: Ingresa valores numéricos válidos.")

    else:
        print("Opción no válida. Intenta de nuevo.")

File: Programas Matlab/Calculadora de Áreas y Perímetros/Calculadora de Áreas y Perímetros.m
% Este programa calcula el área y perímetro de figuras geométricas seleccionadas por el usuario.
% Las opciones son círculo, rectángulo y triángulo (usando fórmula de Herón).

clear; clc; close all;

disp('Calculadora de áreas y perímetros');
disp('1. Círculo');
disp('2. Rectángulo');
disp('3. Triángulo');
opcion = input('Seleccione una figura (1-3): ');

switch opcion
    case 1 % Cálculos para el círculo
        radio = input('Radio del círculo: ');
        area = pi * radio^2;
        perimetro = 2 * pi * radio;
        fprintf('Área: %.2f\nPerímetro: %.2f\n', area, perimetro);
    case 2 % Cálculos para el rectángulo
        largo = input('Largo del rectángulo: ');
        ancho = input('Ancho del rectángulo: ');
        area = largo * ancho;
        perimetro = 2 * (largo + ancho);
        fprintf('Área: %.2f\nPerímetro: %.2f\n', area, perimetro);
    case 3 % Cálculos para el triángulo con fórmula de Herón
        a = input('Lado a del triángulo: ');
        b = input('Lado b del triángulo: ');
        c = input('Lado c del triángulo: ');
        perimetro = a + b + c;
        s = perimetro / 2; % Semiperímetro
        area = sqrt(s * (s - a) * (s - b) * (s - c));
        fprintf('Área: %.2f\nPerímetro: %.2f\n', area, perimetro);
    otherwise
        error('Opción no válida. Seleccione 1, 2 o 3.');
end